src/lib.rs

```diff
@@ -620,6 +620,86 @@ impl<R: RedgeContainers> Redge<R> {
         vn
     }
 
+    /// Only works for triangular faces. It will create a new vertex at the center
```

Comment:
Can a check be added to return an error if the face isn’t triangular?

Reply:
it can but it would be adding an O(n) overhead where n is the valence of the face each time this is called. And since this tends to be called for multiple faces at a time it would be O(f * n) where f is the number of faces and n the largest face valence.

Even for a triangular mesh it's not a small cost.

I think it's better to push the responsibility of the check to the user, we should be using the correctness check regularly in our tests anyway, so if we ever do something improper that's what should catch it.